### PR TITLE
Release with github workflow

### PR DESCRIPTION
**Pull request by black7375**
image
Issue: #165

I created a workflow that puts Mac into the release.
You don't have to expose the GitHub key.

Here's how to permanently erase files from the Git History:
```shell
git filter-branch --tree-filter 'rm -rf ./release-build.sh' HEAD
git filter-branch --prune-empty HEAD
git push origin --force master
```

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        target: ["bash", "fish", "zsh"]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Copy completions
+        run: |
+          cp ${{ matrix.target }}_tealdeer completions_${{ matrix.target }}
+      - uses: softprops/action-gh-release@v1 # actions/create-release, actions/upload-release-asset is deprecated
```
Comment (dbrgn) [inline .github/workflows/release.yml:17]: Huh, it's a bit disappointing that there's no official action for this type of thing 😕

**Review by dbrgn — COMMENTED** [review]
Thanks @black7375 for this config! Hm, in general this looks solid, and it would be nice if we could build our releases in CI. However, for binary builds that we provide to our users as-is, I'm a bit wary of using third party actions. The thing is that we're a single "docker push" away from our next release being compromised.

@black7375 are you aware of other approaches for uploading releases to GitHub in CI? Maybe there's some kind of bash script that does the same thing, which we could check into this repository after a quick review?

**Comment by black7375** [comment]
I'll check it out in the evening

**Comment by black7375** [comment]
@dbrgn

https://gist.github.com/schell/2fe896953b6728cc3c5d8d5f9f3a17a3
I haven't tested it yet, but it looks good.

File: .github/workflows/release.yml (hunk 1 above)
Comment (black7375) [reply .github/workflows/release.yml:17]: I was also embarrassed because it deprecated suddenly.
It's still an incomprehensible decision.

**Comment by dbrgn** [comment]
That looks pretty nice and maintainable!

**Comment by dbrgn** [comment]
Thanks for the update!

I'm not aware of any changes, but this seems to be an OpenSSL linking issue. OpenSSL will get thrown out by #187, so it's best if we wait until that's merged before finishing this.

**Comment by dbrgn** [comment]
@black7375 PR #187 is now merged, OpenSSL is no longer a dependency. Could you rebase this PR?

**Comment by black7375** [comment]
Successfully executed fetch.

**Comment by dbrgn** [comment]
@black7375 what does that mean? 🙂 you did not push any changes.

**Comment by black7375** [comment]
I ran a fetch updstream on GitHub.
Ah. Did you mean to rebase the commit log?

**Comment by dbrgn** [comment]
Yes, it would be great if you could rebase your master branch against the upstream master branch and push the changes (you'll need `git push --force`). Then, you could push a new tag to your fork, to test whether the builds now succeed. The last few workflows all failed: https://github.com/black7375/tealdeer/actions

If you need help with rebasing, just let me know, happy to help!

**Comment by black7375** [comment]
Yes. I rebase and it all passes.

I don't know why the existing action failed.
I have created a separate release-related action, which shouldn't affect the test.

**Comment by dbrgn** [comment]
Replaced by #240.

**Pull request closed without merging**